Modules/_sre.c

```diff
@@ -1414,12 +1414,16 @@ _sre_compile_impl(PyObject *module, PyObject *pattern, int flags,
         self->code[i] = (SRE_CODE) value;
         if ((unsigned long) self->code[i] != value) {
             PyErr_SetString(PyExc_OverflowError,
-                            "regular expression code size limit exceeded");
+                            "regular expression code out of range");
```

Comment:
indeed, but what do you suggest to change?
ISTM that if we just remove the call inside the loop to PyErr_SetString(), a regex code such as
`1 << 32` won't cause an OverflowError on platforms where `sizeof(long) == 8`.